### PR TITLE
Add missing error handling for Ably

### PR DESCRIPTION
**Pull request by KacperKluka**
I've checked all our Ably calls and wrapped them with `try/catch` and added exceptions info to the API docs. There are a few places in which I'm not sure what to do, I've marked them as `TODO` and hope to discuss them in this PR :wink:

### CODE REVIEW

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt [hunk 1]
```diff
@@ -174,9 +171,13 @@ constructor(
...
                             ) {
-                                state.lastSentEnhancedLocations[trackable.id] = event.locationUpdate.location
-                                ably.sendEnhancedLocation(trackable.id, event.locationUpdate)
-                                updateAssetStatuses(state, trackable.id)
+                                try {
+                                    ably.sendEnhancedLocation(trackable.id, event.locationUpdate)
+                                    state.lastSentEnhancedLocations[trackable.id] = event.locationUpdate.location
+                                    updateAssetStatuses(state, trackable.id)
+                                } catch (exception: AblyException) {
+                                    // TODO - what to do here if sending enhanced location fails?
```
Comment (QuintinWillison) [inline publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt:179]: Please create an issue for this.

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt [hunk 2]
```diff
@@ -353,7 +354,11 @@ constructor(
         ably.connect(trackable.id, state.presenceData) { result ->
             try {
                 result.getOrThrow()
-                ably.subscribeForPresenceMessages(trackable.id) { enqueue(PresenceMessageEvent(trackable, it)) }
+                try {
+                    ably.subscribeForPresenceMessages(trackable.id) { enqueue(PresenceMessageEvent(trackable, it)) }
+                } catch (exception: AblyException) {
+                    // TODO - what to do here? should we fail the whole process when subscribing for presence fails? or should it continue?
```
Comment (QuintinWillison) [inline publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt:360]: Please create an issue for this.

File: subscribing-sdk/src/main/java/com/ably/tracking/subscriber/CoreSubscriber.kt [hunk 3]
```diff
@@ -78,8 +78,9 @@ private class DefaultCoreSubscriber(
                         notifyAssetIsOffline()
                         subscribeForEnhancedEvents()
                         subscribeForPresenceMessages()
-                        // TODO - listen for the response
-                        ably.connect(presenceData)
+                        ably.connect(presenceData) {
+                            // TODO what should we do when connection fails?
```
Comment (QuintinWillison) [inline subscribing-sdk/src/main/java/com/ably/tracking/subscriber/CoreSubscriber.kt:82]: Please create an issue for this.

**Review by QuintinWillison — APPROVED** [review]
Thanks, @KacperKluka. This moves things forwards.

For the `TODO` comments which you've added, please create discrete issues for each one. Ideally linking each back to the corresponding comment I've made in this pull request.

At the point you write each issue, please add any thoughts you have as to what might be the right approach to handling the failure along with any mitigations you're aware of which could help in assessing how quickly we decide to fix.

It's probably the case that all of them deserve the [**bug** label](https://github.com/ably/ably-asset-tracking-android/labels/bug).

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt (hunk 1 above)
Comment (KacperKluka) [reply publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt:179]: https://github.com/ably/ably-asset-tracking-android/issues/213

File: publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt (hunk 2 above)
Comment (KacperKluka) [reply publishing-sdk/src/main/java/com/ably/tracking/publisher/CorePublisher.kt:360]: https://github.com/ably/ably-asset-tracking-android/issues/214

File: subscribing-sdk/src/main/java/com/ably/tracking/subscriber/CoreSubscriber.kt (hunk 3 above)
Comment (KacperKluka) [reply subscribing-sdk/src/main/java/com/ably/tracking/subscriber/CoreSubscriber.kt:82]: https://github.com/ably/ably-asset-tracking-android/issues/215

**Pull request merged**